Contributing.md

```diff
@@ -0,0 +1,136 @@
+
+
+## File naming convention
+- File Naming Convention taken from [angular's file naming convention](https://angular.io/guide/styleguide#separate-file-names-with-dots-and-dashes). Loopback internally uses this so this decision is taken to have consistency in file names. This states we use a file name to also represent what it does or is.
+    - an example is "user.model.ts" here we can see the file is a "model" and the name of the model is "user".
+## Api Spec
+- We clearly define our api spec before coding the models and endpoints for them. Please have a look at how to [import api spec in loopback 4.](https://loopback.io/doc/en/lb4/OpenAPI-generator.html)
+
+
+## [Directory structure](https://loopback.io/doc/en/lb4/Loopback-application-layout.html)
+Following the [loopback4 directory structure](https://loopback.io/doc/en/lb4/Loopback-application-layout.html) we have the following directories and their purpose.
+- Under .src we have 5 directories.
+    - __controllers__ : These contain the api endpoints and their request/response structures.
+    - __datasources__ : Contains the datasources used in the codebase.
+    - __models__ : Contains all the data models used across the whole codebase.
+    - __repositories__: Contains the implementations of each model. The database crud and other methods.
+    - __ __tests__ __: Contains the tests for each model (unit , acceptence, integration)
+        - unit
+        - acceptence
+        - integration
+
+We will follow the same structure for our codebase.
+```
+├── src
+│   ├── controllers
+│   │   ├── index.ts
+│   │   └── restaurant.controller.ts
+│   ├── datasources
+│   │   ├── index.ts
+│   │   └── pg.datasource.ts
+│   ├── models
+│   │   ├── index.ts
+│   │   ├── location.model.ts
+│   │   └── restaurant.model.ts
+│   ├── repositories
+│   │   ├── README.md
+│   │   ├── index.ts
+│   │   └── restaurant.repository.ts
+│   └── sequence.ts
+│   ├── __tests__
+│   │   ├── README.md
+│   │   └── acceptance
+|   |       ├── restaurant.controller.acceptance.ts
+│   │       └── test-helper.ts
+│   │   └── unit
+|   |       ├── restaurant.controller.unit.ts
+│   │       └── test-helper.ts
+│   │   └── integration
+|   |       ├── restaurant.integration.unit.ts
+│   │       └── test-helper.ts
+├── public
+│   └── index.html
+├── Dockerfile
+├── README.md
+├── coverage
+├── dist
+├── package.json
+├── tsconfig.json
+├── tsconfig.tsbuildinfo
+├── yarn-error.log
+├── Contribution-guidelines.md
+└── yarn.lock
+```
+
+
+
+## Linting and Formatting
+ Apart from the typescript [Style](https://github.com/microsoft/TypeScript/wiki/Coding-guidelines#style) we use prettier and ESLint.
+- Use [ESLint](https://eslint.org/) to statically analyse the source code and detect common problems. Config used from loopback/eslint-config.
+- [Prettier](https://prettier.io/) to keep our code always formatted the same way. Config stored in .prettierrc
+- Both of these can be done with `  npm run lint:fix  `
```

Comment:
`lint:fix` only fixes lint errors not formatting errors.

Reply:
this does both. yarn run eslint:fix && yarn run prettier:fix.

Reply:
We need a  command to check linting and prettier also. Besides fix.

Reply:
We need a  command to check linting and prettier also. Besides fix.

Reply:
That is already mentioned on line 108 just below this lint:fix.